pytest_splunk_addon/standard_lib/addon_parser/savedsearches_parser.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+Provides savedsearches.conf parsing mechanism
+"""
+
+class SavedSearchParser(object):
+    """
+    Parses savedsearches.conf and extracts savedsearches  
+
+    Args:
+        splunk_app_path (str): Path of the Splunk app
+        app (splunk_appinspect.App): Object of Splunk app
+    """
+    def __init__(self, splunk_app_path, app):
+        self.app = app 
+        self.splunk_app_path = splunk_app_path
+        self._savedsearches = None
+
+    @property
+    def savedsearches(self):
+        try:
+            if not self._savedsearches:
+                self._savedsearches = self.app.get_config("savedsearches.conf")
+            return self._savedsearches
+        except OSError:
+            return None
+
+    def get_savedsearches(self):
+        """
+        Parse the App configuration files & yield savedsearches
+        
+        Yields:
+            generator of list of savedsearches
+        """
+        if not self.savedsearches:
+            return None
+        for stanza in self.savedsearches.sects:
+            savedsearch_sections = self.savedsearches.sects[stanza]
+            savedsearch_container = {
+                        "stanza" : stanza,
+                        "dispatch.earliest_time" : "0",
+                        "dispatch.latest_time" : "now"}
+            for key in savedsearch_sections.options:
+                if key == "search":

```

Comment:
what happens if search param is not provided in the stanza? the whole execution would stop with an exception.

Reply:
Updated code to handle such scenarios and also updated tests in TA_Fiction and TA_Broken accordingly.